Add 'model_generate' programmatic interface (API)

We will create the necessary api to allow users to generate an svm model (in a NoSQL cache), via a programmatic-interface, instead of a standalone web HTML5 interface.

We run our defined unit-test in `pytest_session.py`, from a dedicated terminal:

```bash
vagrant@vagrant-ubuntu-trusty-64:/vagrant/test$ py.test
============================= test session starts ==============================

platform linux2 -- Python 2.7.6, pytest-2.8.0, py-1.4.30, pluggy-0.3.1
rootdir: /vagrant/test, inifile: pytest.ini
collected 4 items

programmatic_interface/pytest_session.py ...F

=================================== FAILURES ===================================

_____________________________ check_model_predict ______________________________


    def check_model_predict():
        """@check_model_predict

        This method tests the 'model_predict' session.

        """

>       assert requests.post(endpoint_url, headers=headers, data=get_sample_json
('sample-model-predict.json'))
E       assert <Response [500]>
E        +  where <Response [500]> = <function post at 0x7f4bf5629050>('http://localhost:5000/load-data/', headers={'Content-Type': 'application/json'}, data='{"properties": {"svm_model_id": "1", "prediction_input[]": ["22.22", "96.24", "338", "72.55", "0.001", "28", "0.678"], "svm_session": "model_predict"}}')
E        +    where <function post at 0x7f4bf5629050> = requests.post
E        +    and   '{"properties": {"svm_model_id": "1", "prediction_input[]":
["22.22", "96.24", "338", "72.55", "0.001", "28", "0.678"], "svm_session": "model_predict"}}' = get_sample_json('sample-model-predict.json')

programmatic_interface/pytest_session.py:82: AssertionError
====================== 1 failed, 3 passed in 0.87 seconds ======================
```

then, we inspect the corresponding output, on the second terminal window, manually running `app.py`:

```bash
127.0.0.1 - - [08/Oct/2015 16:56:35] "POST /load-data/ HTTP/1.1" 500 -
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/flask/app.py", line 1836, in __call__
    return self.wsgi_app(environ, start_response)
  File "/usr/local/lib/python2.7/dist-packages/flask/app.py", line 1820, in wsgi_app
    response = self.make_response(self.handle_exception(e))
  File "/usr/local/lib/python2.7/dist-packages/flask/app.py", line 1403, in handle_exception
    reraise(exc_type, exc_value, tb)
  File "/usr/local/lib/python2.7/dist-packages/flask/app.py", line 1817, in wsgi_app
    response = self.full_dispatch_request()
  File "/usr/local/lib/python2.7/dist-packages/flask/app.py", line 1478, in full_dispatch_request
    response = self.make_response(rv)
  File "/usr/local/lib/python2.7/dist-packages/flask/app.py", line 1577, in make_response
    rv = self.response_class.force_type(rv, request.environ)
  File "/usr/local/lib/python2.7/dist-packages/werkzeug/wrappers.py", line 841, in force_type
    response = BaseResponse(*_run_wsgi_app(response, environ))
  File "/usr/local/lib/python2.7/dist-packages/werkzeug/wrappers.py", line 57, in _run_wsgi_app
    return _run_wsgi_app(*args)
  File "/usr/local/lib/python2.7/dist-packages/werkzeug/test.py", line 867, in run_wsgi_app
    app_rv = app(environ, start_response)
TypeError: 'dict' object is not callable
```

--- a/interface/views.py
+++ b/interface/views.py
@@ -48,7 +48,7 @@ def load_data():
             else: response = loader.get_errors()
 
             # return response
-            return response
+            return json.dumps(response)
 
         # load web-interface
         else:

--- a/test/programmatic_interface/pytest_session.py
+++ b/test/programmatic_interface/pytest_session.py
@@ -71,3 +71,12 @@ def check_model_generate():
     """
 
     assert requests.post(endpoint_url, headers=headers, data=get_sample_json('sample-model-generate.json'))
+
+def check_model_predict():
+    """@check_model_predict
+
+    This method tests the 'model_predict' session.
+
+    """
+
+    assert requests.post(endpoint_url, headers=headers, data=get_sample_json('sample-model-predict.json'))

